### PR TITLE
To go ds sslkeys generate

### PR DESCRIPTION
**Pull request by rob05c**
Branched from #2365 - recommend merging that first.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1791/
Test PASSed.

File: traffic_ops/traffic_ops_golang/deliveryservice/gencert.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+	"github.com/apache/incubator-trafficcontrol/lib/go-util"
+)
+
+const NewCertValidDuration = time.Hour * 24 * 365
+
+// GenerateCert generates a key and certificate for serving HTTPS. The generated key is 2048-bit RSA, to match the old Perl code.
+// The certificate will be valid for NewCertValidDuration time after now.
+// Returns PEM-encoded certificate signing request (csr), certificate (crt), and key; or any error.
+func GenerateCert(host, country, city, state, org, unit string) ([]byte, []byte, []byte, error) {
```
[the remark this answers is not in the document]
Comment (rob05c) [reply traffic_ops/traffic_ops_golang/deliveryservice/gencert.go]: I don't agree, IMO named returns are inherently less readable. They're very uncommon in languages, they're unexpected, it's confusing when people expect variables to be declared locally or in parameters, it's confusing when the variable is actually created and initialized by the runtime, and it's confusing how they act in defer statements.

Sure, you can memorize the Go language behavior in all those cases. But none of them are intuitive or obvious.

Moreover, the Go community generally agrees, and more projects omit than use them.

IMO we should have Go coding standards for trafficcontrol which, among other things, says not to use named returns, for the above reasons.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1809/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1851/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1965/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2015/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2049/
Test PASSed.

**Pull request merged**